build.gradle

```diff
@@ -1,181 +1,252 @@
+                }
+            }
+            // Local testing maven repo
+            maven {
+                name = "Testing"
+                url = "${rootProject.projectDir}/build/testing-mvn-repo"
+            }
+        }
+    }
 }
 
-group = 'com.dumbdogdiner'
+subprojects {
+    apply plugin: "io.freefair.lombok"
+    apply plugin: "com.github.hierynomus.license"
+
+    // License Plugin Options
+    license {
+        header = project.rootProject.file("LICENSE_HEADER")
+        ext.year = Calendar.getInstance().get(Calendar.YEAR)
+        mapping("java", "SLASHSTAR_STYLE")
+        exclude "**/*.json" // Exclude JSON to keep the font width data valid
+        exclude "**/*.md"
+        exclude "**/*.MockMaker"
+    }
 
-version = '2.1.0'
+    tasks.withType(JavaCompile) {
+        options.compilerArgs << "-Xlint:unchecked" << "-Xlint:deprecation" << "-XDignore.symbol.file"
+    }
 
-// License Plugin Options
-license {
-    header = project.file('LICENSE_HEADER')
-    ext.year = Calendar.getInstance().get(Calendar.YEAR)
-    mapping("java", "SLASHSTAR_STYLE")
-    exclude "**/*.json" // Exclude JSON to keep the font width data valid
-}
+    configurations {
+        jaxDoclet
+        // give test dependencies access to compileOnly dependencies to emulate providedCompile
+        testImplementation.extendsFrom compileOnly
+    }
 
-tasks.withType(JavaCompile) {
-    options.compilerArgs << "-Xlint:unchecked" << "-Xlint:deprecation" << "-XDignore.symbol.file"
-}
+    repositories {
+        // Note: Maven Central is defined in allprojects
+        jcenter()
+
+        maven { url "https://oss.sonatype.org/content/repositories/snapshots" }
+        maven { url "https://papermc.io/repo/repository/maven-public/" }
+
+        // Note: Subprojects can't have repo overrides to we have to put it here
+        // Define a Ivy repo for the font width data (that way we don't need another plugin!)
+        def ddd_mc_font = ivy {
+           url "https://dumbdogdiner.github.io/"
+            patternLayout { artifact "/[module]/[revision]/[classifier].[ext]"}
+            metadataSources { artifact() } 
+        }
+        // Only use the Ivy repo for font width data - speeds up dependency resolution
+        exclusiveContent {
+            forRepositories(ddd_mc_font)
+            filter { includeGroup("dumbdogdiner") }
+        }
+    }
 
-configurations {
-    jaxDoclet
-    // give test dependencies access to compileOnly dependencies to emulate providedCompile
-    testImplementation.extendsFrom compileOnly
-}
+    dependencies {
+        compileOnly "org.projectlombok:lombok:1.18.18"
+        annotationProcessor "org.projectlombok:lombok:1.18.18"
 
-repositories {
-    mavenCentral()
-    jcenter()
+        // Note: Common uses both of these... ???
+        compileOnly "net.md-5:bungeecord-api:1.16-R0.5-SNAPSHOT"
+        compileOnly "com.destroystokyo.paper:paper-api:1.16.5-R0.1-SNAPSHOT"
```

Comment:
Modules should implement these on their own, should they not? This would give `bukkit` access to Bungeecord and vice versa....????? Ideally common shouldn't hard-depend on either and only use deps common between the two e.g. `ChatColor`

Reply:
Blocked by #98.

Reply:
Confusion?